Vloginov/add polynomials handwritten (#1713)
openvinotoolkit/open_model_zoo: models/intel/formula-recognition-polynomials-handwritten-0001/description/formula-recognition-polynomials-handwritten-0001.md

# formula-recognition-polynomials-handwritten-0001 (composite)

## Use Case and High-Level Description

This is an im2latex composite model that recognizes latex formulas.
The model uses vocabulary file to predict sequence of latex tokens.

> The model is built on the resnext50 backbone with additional attention-based text recognition head.

Reviewer: would be nice to add what dataset model was trained on. If that is internal dataset then just say trained on internal dataset, containing handwritten symbols

Author: Thank you for your suggestion, the description was updated.

## Example of the input data

![](./formula-recognition-polynomials-handwritten-0001.png)

## Example of the output

`- 4 . 6 c ^ { 2 } d ^ { - 6 0 }`

## Composite model specification

| Metric                                        | Value     |
|-----------------------------------------------|-----------|
| im2latex_polynomials_handwritten dataset, im2latex-match-images metric | 70.5% |
| Source framework                              | PyTorch\* |

Im2latex-match-images metric is calculated by [this algorithm](../../../../tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py )

## Encoder model specification

The formula-recognition-polynomials-handwritten-0001-encoder model is a ResNeXt-50 like backbone with some initialization layers for decoder

| Metric                                        | Value     |
|-----------------------------------------------|-----------|
| GFlops                                        | 12.8447   |
| MParams                                       | 8.6838    |

### Performance

### Inputs

> 1.	Name: `imgs` , shape: [1x3x100x990]. An input image in the [1xCxHxW] format.

Reviewer: ```suggestion
1.	Name: `imgs` , shape: [1x3x96x990]. An input image in the [1xCxHxW] format.
```

    The expected channel order is BGR.

### Outputs
1.	Name: `hidden`, shape: [1x512]. Initial context state of the LSTM cell.
2.	Name: `context`, shape: [1x512]. Initial hidden state of the LSTM cell.
3.	Name: `init_0`, shape: [1x256]. Initial state of the decoder.

> 5.	Name: `row_enc_out`, shape: [1x7x62x512]. Features from encoder that are fed to a decoder.

Reviewer: ```suggestion
4.	Name: `row_enc_out`, shape: [1x7x62x512]. Features from encoder that are fed to a decoder.
```

## Decoder model specification

The formula-recognition-polynomials-handwritten-0001-decoder model is an LSTM based decoder with attention module.

| Metric                                        | Value     |
|-----------------------------------------------|-----------|
| GFlops                                        | 0.2017    |
| MParams                                       | 2.5449    |

### Performance

### Inputs

1.	Name: `dec_st_c` , shape: [1x512]. Current context state of the LSTM cell.
2.	Name: `dec_st_h` , shape: [1x512]. Current hidden state of the LSTM cell.
3.	Name: `output_prev`, shape: [1x256]. Current state of the decoder.
4.	Name: `row_enc_out`, shape: [1x7x62x512]. Encoded features.
5.	Name: `tgt`, shape: [1x1]. Index of the previous symbol.

### Outputs

1.	Name: `dec_st_c` , shape: [1x512]. Current context state of the LSTM cell.
2.	Name: `dec_st_h` , shape: [1x512]. Current hidden state of the LSTM cell.
3.	Name: `output`, shape: [1x256]. Current state of the decoder.

> 1.	Name: `logit`, shape: [1xVocab_Size]. Classification confidence scores in the [0, 1] range

Reviewer: ```suggestion
4.	Name: `logit`, shape: [1xVocab_Size]. Classification confidence scores in the [0, 1] range
```

3.	Name: `init_0`, shape: [1x256]. Initial state of the decoder.
4.	Name: `row_enc_out`, shape: [1x6x62x512]. Features from encoder that are fed to a decoder.

## Decoder model specification

The formula-recognition-polynomials-handwritten-0001-decoder model is an LSTM based decoder with attention module.

| Metric                                        | Value     |
|-----------------------------------------------|-----------|
| GFlops                                        | 0.2017    |
| MParams                                       | 2.5449    |

### Performance

### Inputs

1.	Name: `dec_st_c` , shape: [1x512]. Current context state of the LSTM cell.
2.	Name: `dec_st_h` , shape: [1x512]. Current hidden state of the LSTM cell.
3.	Name: `output_prev`, shape: [1x256]. Current state of the decoder.
4.	Name: `row_enc_out`, shape: [1x6x62x512]. Encoded features.
5.	Name: `tgt`, shape: [1x1]. Index of the previous symbol.

### Outputs

1.	Name: `dec_st_c` , shape: [1x512]. Current context state of the LSTM cell.
2.	Name: `dec_st_h` , shape: [1x512]. Current hidden state of the LSTM cell.
3.	Name: `output`, shape: [1x256]. Current state of the decoder.

> 4.	Name: `logit`, shape: [1xVocab_Size]. Classification confidence scores in the [0, 1] range

Reviewer 2: ```suggestion
4.	Name: `logit`, shape: [1xN], where `N` is a vocabulary size. Classification confidence scores in the [0, 1] range
```
